base/strings/string.jl

```diff
@@ -16,19 +16,22 @@ const ByteArray = Union{Vector{UInt8},Vector{Int8}}
 # String constructor docstring from boot.jl, workaround for #16730
 # and the unavailability of @doc in boot.jl context.
 """
-    String(v::Vector{UInt8})
+    String(v::AbstractVector{UInt8})
 
 Create a new `String` from a vector `v` of bytes containing
-UTF-8 encoded characters.   This function takes "ownership" of
-the array, which means that you should not subsequently modify
-`v` (since strings are supposed to be immutable in Julia) for
-as long as the string exists.
+UTF-8 encoded characters.
+
+When possible, the data in `v` will be used directly rather than making
+a copy. This is the case for `Vector{UInt8}` arrays returned by [`take!`](@ref)
+on a writable [`IOBuffer`](@ref) or by [`read(io, nb)`](@ref). Else, a copy is made.
+In all cases, one should assume that the function takes "ownership" of the array
```

Comment:
"In all cases" seems overly broad here.  Why not just say that this is the case for a `Vector` and leave it at that?